src/geometry.cpp

```diff
@@ -0,0 +1,171 @@
+      float height = noise.octaveNoise0_1(((float)i)/((float)depth-1.0), ((float)j)/((float)depth-1.0), octaves) - 0.5;
+      height -= std::pow(
+        std::sqrt(
+          std::pow(-0.5 + ((float)i)*(1.0/((float)depth-1.0)),2) +
+          std::pow(-0.5 + ((float)j)*(1.0/((float)depth-1.0)),2)
+        ) / 0.75,
+        2
+      );
+      heights[heights.size()-1].push_back(height);
+    }
+  }
+
+  for (int i = 0; i < heights.size()-1; i++) {
+    for (int j = 0; j < heights[i].size()-1; j++) {
+      float x1 = -0.5 + ((float)i)*(1.0/((float)depth-1.0));
+      float z1 = -0.5 + ((float)j)*(1.0/((float)depth-1.0));
+      float y1 = heights[i][j];
+
+      float x2 = -0.5 + ((float)(i+1))*(1.0/((float)depth-1.0));
+      float z2 = -0.5 + ((float)j)*(1.0/((float)depth-1.0));
+      float y2 = heights[i+1][j];
+
+      float x3 = -0.5 + ((float)i)*(1.0/((float)depth-1.0));
+      float z3 = -0.5 + ((float)(j+1))*(1.0/((float)depth-1.0));
+      float y3 = heights[i][j+1];
+
+      float x4 = -0.5 + ((float)(i+1))*(1.0/((float)depth-1.0));
+      float z4 = -0.5 + ((float)(j+1))*(1.0/((float)depth-1.0));
+      float y4 = heights[i+1][j+1];
+
+      // Triangle 1
+      vertices.push_back(x1);
+      vertices.push_back(y1);
+      vertices.push_back(z1);
+
+      vertices.push_back(x2);
+      vertices.push_back(y2);
+      vertices.push_back(z2);
+
+      vertices.push_back(x3);
+      vertices.push_back(y3);
+      vertices.push_back(z3);
+
+      // Triangle 2
+      vertices.push_back(x4);
+      vertices.push_back(y4);
+      vertices.push_back(z4);
+
+      vertices.push_back(x3);
+      vertices.push_back(y3);
+      vertices.push_back(z3);
+
+      vertices.push_back(x2);
+      vertices.push_back(y2);
+      vertices.push_back(z2);
+    }
+  }
+
+  return addNormals(vertices);
+}
+
+std::vector<float> cube() {
+  return addNormals({
+      -0.5f, -0.5f, -0.5f,
+       0.5f, -0.5f, -0.5f,
+       0.5f,  0.5f, -0.5f,
+       0.5f,  0.5f, -0.5f,
+      -0.5f,  0.5f, -0.5f,
+      -0.5f, -0.5f, -0.5f,
+
+       0.5f,  0.5f,  0.5f,
+       0.5f, -0.5f,  0.5f,
+      -0.5f, -0.5f,  0.5f,
+      -0.5f, -0.5f,  0.5f,
+      -0.5f,  0.5f,  0.5f,
+       0.5f,  0.5f,  0.5f,
+
+      -0.5f,  0.5f,  0.5f,
+      -0.5f,  0.5f, -0.5f,
+      -0.5f, -0.5f, -0.5f,
+      -0.5f, -0.5f, -0.5f,
+      -0.5f, -0.5f,  0.5f,
+      -0.5f,  0.5f,  0.5f,
+
+       0.5f,  0.5f,  0.5f,
+       0.5f,  0.5f, -0.5f,
+       0.5f, -0.5f, -0.5f,
+       0.5f, -0.5f, -0.5f,
+       0.5f, -0.5f,  0.5f,
+       0.5f,  0.5f,  0.5f,
+
+      -0.5f, -0.5f, -0.5f,
+       0.5f, -0.5f, -0.5f,
+       0.5f, -0.5f,  0.5f,
+       0.5f, -0.5f,  0.5f,
+      -0.5f, -0.5f,  0.5f,
+      -0.5f, -0.5f, -0.5f,
+
+      -0.5f,  0.5f, -0.5f,
+       0.5f,  0.5f, -0.5f,
+       0.5f,  0.5f,  0.5f,
+       0.5f,  0.5f,  0.5f,
+      -0.5f,  0.5f,  0.5f,
+      -0.5f,  0.5f, -0.5f
+  });
+}
+
+std::vector<float> sphere() {
+  static float vertexLocations[] = {
```

Comment:
Let's change these local static variables to be `static constexpr` instead. That way, it will be easy to change these numbers from being hardcoded to, for example, fractions or in terms of other compile-time values.